drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -852,38 +862,56 @@ local function sat_attr_handler(driver, device, ib, response)
 end
 
 local function temp_attr_handler(driver, device, ib, response)
-  if ib.data.value ~= nil then
-    if (ib.data.value < COLOR_TEMPERATURE_MIRED_MIN or ib.data.value > COLOR_TEMPERATURE_MIRED_MAX) then
-      device.log.warn_with({hub_logs = true}, string.format("Device reported color temperature %d mired outside of sane range of %.2f-%.2f", ib.data.value, COLOR_TEMPERATURE_MIRED_MIN, COLOR_TEMPERATURE_MIRED_MAX))
-      return
-    end
-    local temp = utils.round(MIRED_KELVIN_CONVERSION_CONSTANT/ib.data.value)
-    local temp_device = device
-    if device:get_field(IS_PARENT_CHILD_DEVICE) == true then
-      temp_device = find_child(device, ib.endpoint_id) or device
-    end
-    local most_recent_temp = temp_device:get_field(MOST_RECENT_TEMP)
-    -- this is to avoid rounding errors from the round-trip conversion of Kelvin to mireds
-    if most_recent_temp ~= nil and
-      most_recent_temp <= utils.round(MIRED_KELVIN_CONVERSION_CONSTANT/(ib.data.value - 1)) and
-      most_recent_temp >= utils.round(MIRED_KELVIN_CONVERSION_CONSTANT/(ib.data.value + 1)) then
-        temp = most_recent_temp
-    end
-    device:emit_event_for_endpoint(ib.endpoint_id, capabilities.colorTemperature.colorTemperature(temp))
+  local temp_in_mired = ib.data.value
+  if temp_in_mired == nil then
+    return
   end
+  if (temp_in_mired < COLOR_TEMPERATURE_MIRED_MIN or temp_in_mired > COLOR_TEMPERATURE_MIRED_MAX) then
+    device.log.warn_with({hub_logs = true}, string.format("Device reported color temperature %d mired outside of sane range of %.2f-%.2f", temp_in_mired, COLOR_TEMPERATURE_MIRED_MIN, COLOR_TEMPERATURE_MIRED_MAX))
+    return
+  end
+  local min_temp_mired = get_field_for_endpoint(device, COLOR_TEMP_BOUND_RECEIVED_MIRED..COLOR_TEMP_MIN, ib.endpoint_id)
```

Comment:
Are we storing min and max for both kelvin and mired? It seems to me that we should only need to store one and that can be our source of truth. I don't know if this was introduced in this PR or if it already existed, but I think we'd realistically only want to store one of these values and can do our conversions in the driver as needed unless there's a good reason to store both.

Reply:
The reason to store both is that it would allow us to know when we are dealing with a value at the minimum or maximum of the allowed range, either in terms of Kelvin when using the slider in the app, or in terms of mireds when dealing with values received from the device. Like I mentioned in the other comment, if the capability is set to the maximum for example in the plugin, then the driver would just convert this to the nearest value in mireds rather than setting the device to the maximum allowed value in mireds.

Going the other direction, if the device reported that its color temp is equal to the minimum mireds value (153 in the example in the ticket), the driver currently just converts this to kelvin and emits that value, without checking that it is within the allowed kelvin range. 153 mireds = 6535 kelvin, which exceeds the rounded value of 6500 that we calculated for the max kelvin capability. This results in the plugin saying "Out of range"

The changes in this PR would prevent this by just emitting the max kelvin value of 6500 if we receive the minimum allowed mireds value, and if we receive a capability command that is setting the temp to the max of 6500, then we would just send the min allowed mireds value of 153 to the device. (And the same kind of thing if we are dealing with the minimum instead.)